kiss

```diff
@@ -868,7 +868,10 @@ pkg_conflicts() {
     # Store the list of found conflicts in a file as we'll be using the
     # information multiple times. Storing things in the cache dir allows
     # us to be lazy as they'll be automatically removed on script end.
-    grep -Fxf "$mak_dir/$pid-m" -- "$@" 2>/dev/null > "$mak_dir/$pid-c" ||:
+
+    # The first sort is to remove duplicates in existing packages' manifest
+    # due to shared directories.
+    sort -u "$@" | sort "$mak_dir/$pid-m" - | uniq -d > "$mak_dir/$pid-c" ||:
```

Comment:
Instead of `sort -u "$@"` you should use `sed '/\/$/d' "$@"` which should be even faster. This will delete directories from the manifests, and we don't need to wait for `sort` to finish.